.github/actions/compute-matrix/action.yaml

```diff
@@ -0,0 +1,41 @@
+name: Compute matrix
+
+outputs:
+  BUILD_MATRIX:
+    value: ${{ steps.compute-matrix.outputs.BUILD_MATRIX }}
+  TEST_MATRIX:
+    value: ${{ steps.compute-matrix.outputs.TEST_MATRIX }}
+
+runs:
+  using: composite
+  steps:
+    - name: Compute matrix
+      id: compute-matrix
+      shell: bash
+      run: |
+        set -eo pipefail
+
+        export BUILD_MATRIX="
+        - { CUDA_VER: '12.0.1', ARCH: 'amd64', PY_VER: '3.12', LINUX_VER: 'rockylinux8' }
+        "
+
+        export TEST_MATRIX="
```

Comment:
Do we really need 8 jobs? Can we do a reduced matrix here of 3 jobs, and then do the rest with a separate set of nightly jobs?

Reply:
No, if you look at the pr.yaml, there's a matrix filter that only issue 2 jobs for each of the tests. One for amd64, and the other for arm64.